src/internal/connector/exchange/container_resolver.go

```diff
@@ -40,50 +48,147 @@ type containersEnumerator interface {
 // folders if each folder is only a single character.
 const maxIterations = 300
 
-func newContainerResolver() *containerResolver {
+func newContainerResolver(refresher containerRefresher) *containerResolver {
 	return &containerResolver{
-		cache: map[string]graph.CachedContainer{},
+		cache:     map[string]graph.CachedContainer{},
+		refresher: refresher,
 	}
 }
 
 type containerResolver struct {
-	cache map[string]graph.CachedContainer
+	cache     map[string]graph.CachedContainer
+	refresher containerRefresher
 }
 
 func (cr *containerResolver) IDToPath(
 	ctx context.Context,
 	folderID string,
 ) (*path.Builder, *path.Builder, error) {
-	return cr.idToPath(ctx, folderID, 0)
+	ctx = clues.Add(ctx, "container_id", folderID)
+
+	c, ok := cr.cache[folderID]
+	if !ok {
+		return nil, nil, clues.New("folder not cached").WithClues(ctx)
```

Comment:
We swap our vocabulary a lot between container, folder, and sometimes calendar.  That makes sense when we know what the item specifically is (ie: a folder vs a calendar).  In the generic case, do we want to call them `containers` or `folders`?
```suggestion
		return nil, nil, clues.New("container not cached").WithClues(ctx)
```